File: src/scan.js
const {PythonShell} = require('python-shell');
const {program, InvalidArgumentError} = require('commander');
const fs = require('fs');

function getArgs() {
  program
    .description('Scan the text of a document')
    .requiredOption('-i, --input <string>', 'Image path or Data URL', value => {
      const matches = value.match(/^\s*data:(?:(\w+\/[\w\d\-+.]+)(?:;[\w-]+=[\w\d-]+)?)?(?:;base64)?,([\w\d!$&\',()*+;=\-._~:@\/?%\s]*)\s*$/);
      if (matches) {
        const mime = matches[1];
        if (mime !== 'image/png' && mime !== 'image/jpeg')
          throw new InvalidArgumentError('Only jpeg or png can be used for the input DataURL');
      } else if (!fs.existsSync(value) || !fs.lstatSync(value).isFile())
        throw new InvalidArgumentError('File not found');
      return value;
    })
    .requiredOption('-t, --type <string>', 'OCR document type. \'driverslicense\' or \'mynumber\' can be used', value => {
      if (!['driverslicense', 'mynumber'].includes(value))
        throw new InvalidArgumentError('The type can be "driverslicense" or "mynumber"');
      return value;
    })
    .option('-d, --debug', 'Display debug image on display', false)
    .parse();
  return program.opts();
}
const args = getArgs();
const pythonArgs = ['-i', args.img, '-t', args.type];
if (args.debug)
  pythonArgs.push('-d');
let matches = await new Promise((rslv, rej) => {
  PythonShell.run('scan_cli.py', {
    // pythonPath: '/usr/local/bin/python3.9',
    scriptPath: __dirname,
    args: pythonArgs,
    mode: 'text'
  }, (err, res) => {
    if (err)
      return void rej(err);
    rslv(res);
  });
});
console.log(matches);

File: src/scan.py
import os
from dotmap import DotMap
import core
import utils

def scan(img, type, scan_fields = None, debug_img_callback = None):
  """Scan text from a document.
  Args:
      img (str): Image path or DataURL.
      type (str): Document type. (driverslicense: Driver's license card, mynumber: My number card)
      scan_fields (list, optional): Fields to be scanned. Defaults to None(all fields).
      debug_img_callback (function, optional): Callback function for image transformation. Defaults to None.
  Raises:
      ValueError: The img parameter is incorrect.
      ValueError: The type parameter is incorrect.
  """
  try:
    _validate_params(img, type)
    if not debug_img_callback:
      debug_img_callback = lambda label, img: None
    data_url = core.detect_id_card(img, debug_img_callback)
    if not data_url:
      utils.logging.debug('The document could not be detected from the image')
      return DotMap()
    return core.scan_text(data_url, type, scan_fields, debug_img_callback)
  except:
    utils.logging.exception('')
    raise

def _validate_params(img, type):
  if (not utils.is_data_url(img) and not os.path.exists(img) and not os.path.isfile(img)):
    raise ValueError('Input is incorrect. Input can be an image path, or DataURL')
  if type != 'driverslicense' and type != 'mynumber':
    raise ValueError('Incorrect type. Type can be "driverslicense" or "mynumber"')